subprojects/version-control/src/main/java/org/gradle/vcs/internal/DefaultVersionControlSystemFactory.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.vcs.internal;
+
+import org.gradle.vcs.VersionControlSpec;
+import org.gradle.vcs.VersionControlSystem;
+import org.gradle.vcs.git.internal.GitVersionControlSystem;
+
+public class DefaultVersionControlSystemFactory implements VersionControlSystemFactory {
+    @Override
+    public VersionControlSystem create(VersionControlSpec spec) {
+        // TODO: Register these mappings somewhere
+        if (spec instanceof DirectoryRepository) {
+            return new SimpleVersionControlSystem();
+        }
```

Comment:
We should explicitly check for `GitVersionControlSpec` as well. Let the last line return null or throw an appropriate runtime exception.

Reply:
Yep, we should explode if we don't understand the spec type